main.go

```diff
@@ -29,194 +29,215 @@ import (
-				}
+			err := cmd.Process.Signal(sig)
+			if err != nil {
+				slog.Warn(
+					fmt.Errorf("failed to signal process: %w", err).Error(),
+					slog.String("signal", sig.String()),
+				)
 			}
-		}()
+		}
+	}()
 
-		err = cmd.Wait()
+	err = cmd.Wait()
 
-		close(sigs)
+	close(sigs)
 
-		if err != nil {
-			exitCode := -1
-			// try to get the original exit code if possible
-			var exitError *exec.ExitError
-			if errors.As(err, &exitError) {
-				exitCode = exitError.ExitCode()
-			}
+	if err != nil {
+		slog.Error(fmt.Errorf("failed to exec process: %w", err).Error())
 
-			logger.Error(fmt.Errorf("failed to exec process: %w", err).Error(), slog.String("entrypoint", fmt.Sprint(entrypointCmd)))
+		// Exit with the original exit code if possible
+		var exitErr *exec.ExitError
+		if errors.As(err, &exitErr) {
+			os.Exit(exitErr.ExitCode())
+		}
+
+		os.Exit(-1)
+	}
+
+	os.Exit(cmd.ProcessState.ExitCode())
+}
+
+func initLogger(config *config.Config) {
+	var level slog.Level
+
+	err := level.UnmarshalText([]byte(config.LogLevel))
+	if err != nil { // Silently fall back to info level
+		level = slog.LevelInfo
+	}
 
-			os.Exit(exitCode)
+	levelFilter := func(levels ...slog.Level) func(ctx context.Context, r slog.Record) bool {
+		return func(ctx context.Context, r slog.Record) bool {
+			return slices.Contains(levels, r.Level)
 		}
+	}
 
-		os.Exit(cmd.ProcessState.ExitCode())
+	router := slogmulti.Router()
+
+	if config.JSONLog {
+		// Send logs with level higher than warning to stderr
+		router = router.Add(
+			slog.NewJSONHandler(os.Stderr, &slog.HandlerOptions{Level: slog.LevelWarn}),
+			levelFilter(slog.LevelWarn, slog.LevelError),
+		)
+
+		// Send info and debug logs to stdout
+		router = router.Add(
+			slog.NewJSONHandler(os.Stdout, &slog.HandlerOptions{Level: slog.LevelDebug}),
+			levelFilter(slog.LevelDebug, slog.LevelInfo),
+		)
+	} else {
+		// Send logs with level higher than warning to stderr
+		router = router.Add(
+			slog.NewTextHandler(os.Stderr, &slog.HandlerOptions{Level: slog.LevelWarn}),
+			levelFilter(slog.LevelWarn, slog.LevelError),
+		)
+
+		// Send info and debug logs to stdout
+		router = router.Add(
+			slog.NewTextHandler(os.Stdout, &slog.HandlerOptions{Level: slog.LevelDebug}),
+			levelFilter(slog.LevelDebug, slog.LevelInfo),
+		)
 	}
-	err = syscall.Exec(binary, entrypointCmd, secretsEnv)
-	if err != nil {
-		logger.Error(fmt.Errorf("failed to exec process: %w", err).Error(), slog.String("entrypoint", fmt.Sprint(entrypointCmd)))
 
-		os.Exit(1)
+	if config.LogServer != "" {
+		writer, err := net.Dial("udp", config.LogServer)
+
+		// We silently ignore syslog connection errors for the lack of a better solution
+		if err == nil {
+			router = router.Add(slogsyslog.Option{Level: slog.LevelInfo, Writer: writer}.NewSyslogHandler())
+		}
 	}
+
+	// TODO: add level filter handler
+	logger := slog.New(router.Handler())
+	logger = logger.With(slog.String("app", "vault-secret-init"))
+
+	slog.SetDefault(logger)
```

Comment:
indicate here that the default logger is set and `slog` directly can be used to access it